cargo-near-build/src/near/build/mod.rs

```diff
@@ -244,27 +270,56 @@ fn is_newer_than(prev: &Utf8PathBuf, next: &Utf8PathBuf) -> bool {
     prev_time > next_time
 }
 
+const MAX_VERSION_NO_BULK_MEMORY: rustc_version::Version = rustc_version::Version::new(1, 86, 0);
+const MIN_VERSION_WITH_BULK_MEMORY_NTRAPPING_FLOAT_TO_INT: rustc_version::Version =
+    rustc_version::Version::new(1, 87, 0);
 fn maybe_wasm_opt_step(
     input_path: &Utf8PathBuf,
     no_wasmopt: bool,
+    rustc_version: &rustc_version::Version,
 ) -> eyre::Result<(Utf8PathBuf, Option<NamedTempFile>)> {
     let result = if !no_wasmopt {
         let opt_destination = tempfile::Builder::new()
             .prefix("optimized-")
             .suffix(".wasm")
             .tempfile()?;
         println!();
+        let additional_features = {
+            let mut features = vec![];
+            if *rustc_version >= MIN_VERSION_WITH_BULK_MEMORY_NTRAPPING_FLOAT_TO_INT {
```

Comment:
I'm just interested, why don't we just fail building instead of compiling it anyway?

Reply:
well to have smth to test with when or if the features are supported by nearcore. 
And to have a correct reason of failure in user reports). The root cause of a problem here isn't a choke by cargo near. 
i needed to test earlier rather than later if current `wasm-opt-rs` is capable to chew these new features. There's an update pr to 121 version binaryen, but it doesn't look it will be finalized soon.

Reply:
makes sense.